src/cartridge.rs

```diff
@@ -0,0 +1,78 @@
+use std::{
+    fs::File,
+    io::{self, Read},
+    rc::Rc,
+};
+
+///Contains all the information about the header of the cartridge
+pub(crate) struct CartridgeHeader {
+    title: String,
+    //WIP, missing properties
+}
+
+impl CartridgeHeader {
+    pub fn new(data: &[u8]) -> Self {
+        Self::verify_checksum(data).expect("Invalid checksum");
+        let title = String::from_utf8(data[0x00A0..0x00AC].to_vec()).expect("Invalid UTF-8 title");
+        if !title.is_ascii() {
+            panic!("Expected an ascii sequence");
+        }
+
+        /*fetch other data here*/
+
+        println!("Title = {title}");
+
+        Self { title }
+    }
+
+    pub(crate) fn title(&self) -> &str {
+        self.title.as_str()
+    }
+
+    ///Verifies the header checksum
```

Comment:
Is this comment important? As I wrote before, explain why checksum is important things to verify or other similar but just rewrite name of the function is only adding cognitive load.